src/Microsoft.PowerShell.ConsoleHost/host/msh/ProgressNode.cs

```diff
@@ -337,6 +357,82 @@ private static void RenderFullDescription(string description, string indent, int
                     maxWidth));
         }
 
+        /// <summary>
+        /// Renders a node in the "ANSI" style.
+        /// </summary>
+        /// <param name="strCollection">
+        /// List of strings to which the node's rendering will be appended.
+        /// </param>
+        /// <param name="indentation">
+        /// The indentation level in chars at which the node should be rendered.
+        /// </param>
+        /// <param name="maxWidth">
+        /// The maximum number of chars that the rendering is allowed to consume.
+        /// </param>
+        private
+        void
+        RenderAnsi(ArrayList strCollection, int indentation, int maxWidth)
```

Comment:
The method uses so many string operations that we could use Span overloads not to be so expensive.

Reply:
I'm using StringBuilder which internally uses spans already

Reply:
I don't see you use StringBuilder span overloads.
And it seems we could exclude StringBuilder at all.

Reply:
I looked at creating a fix span to render the output, but the code ended up much harder to read than using StringBuilder and it's not clear to me of getting much benefit as the string parts that needed to be appended still need to be allocated even if copying spans around.  Also the span overloads for StringBuilder don't help here as the appended strings are initially strings.